envoy/network/address.h

```diff
@@ -230,6 +231,39 @@ class Instance {
    * @return SocketInterface to be used with the address.
    */
   virtual const Network::SocketInterface& socketInterface() const PURE;
+
+  /**
+   * For support the absl::Hash.
+   */
+  virtual void abslHashValue(absl::HashState state) const PURE;
```

Comment:
IMHO we should support hashing directly without the indirect `AddressKey`. None of the current code hashes directly, and using structured hashing and equality is strictly better long term.

Reply:
This is due to I want to hash the shared pointer of address instance object `InstanceConstSharedPtr `, otherwise I thought it should expensive to store each address instance object copy in the hash map.

Reply:
OK, that makes sense.